libcudacxx/docs/extended_api/synchronization_primitives/barrier/arrive_tx.md

```diff
@@ -0,0 +1,66 @@
+---
+grand_parent: Extended API
+parent: Barriers
+---
+
+# `cuda::device::arrive_tx`
+
+Defined in header `<cuda/barrier>`:
+
+```cuda
+__device__
+cuda::barrier<cuda::thread_scope_block>::arrival_token
```

Comment:
This needs to be a template to support non-empty completion functions.
That has to be done now, since it would be an ABI breaking change otherwise.